src/software/thunderscope/gl/layers/gl_referee_info_layer.py

```diff
@@ -0,0 +1,249 @@
+import math
+import time
+from typing import Optional
+
+from PyQt6 import QtGui
+from google.protobuf.json_format import MessageToDict
+from pyqtgraph.opengl.items.GLTextItem import GLTextItem
+
+from proto.import_all_protos import *
+from software.py_constants import *
+from software.thunderscope.constants import DepthValues, Colors
+from software.thunderscope.gl.graphics.gl_circle import GLCircle
+from software.thunderscope.gl.graphics.gl_label import GLLabel
+from software.thunderscope.gl.helpers.observable_list import ObservableList
+from software.thunderscope.gl.layers.gl_layer import GLLayer
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+
+
+class GLRefereeInfoLayer(GLLayer):
+    """GLLayer that visualizes referee info"""
+
+    REFEREE_COMMAND_PREFIX = "Command: "
+    GAMESTATE_PREFIX = "Game State: "
+
+    # outline color of the avoid area for ball placement
+    BALL_PLACEMENT_ROBOT_AVOID_AREA_VISUALIZATION_COLOR = Colors.RED
+    # outline color of the tolerance region for ball placement (when the ball is inside)
+    BALL_PLACEMENT_TOLERANCE_VISUALIZATION_COLOR_ON = Colors.GREEN
+    # outline color of the tolerance region for ball placement (when the ball is outside)
+    BALL_PLACEMENT_TOLERANCE_VISUALIZATION_COLOR_OFF = Colors.RED
+    # outline color of the target mark
+    BALL_PLACEMENT_TARGET_VISUALIZATION_COLOR = Colors.FIELD_LINE_COLOR
+    # text color for count down
+    COUNT_DOWN_TEXT_COLOR = Colors.RED
+
+    @staticmethod
+    def is_point_in_circle(
+        point: tuple[float | int, float | int],
+        center: tuple[float | int, float | int],
+        radius: float | int,
+    ) -> bool:
+        """Returns true if the point is in the circle.
```

Comment:
nit: wouldn't it be better for this method to take in 2 points instead? it looks like where you're using it, you're manually converting Point objects to this tuple. Most places in our code deal with Points instead of tuples so would be useful for that as well. Also would make the method signature a bit cleaner lol